### PR TITLE
ISPN-12424 Support PEM files via the Hot Rod client trustStoreFileNam…

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-12427

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/proc_configure_clients_trust_server.adoc [hunk 1]
```diff
@@ -13,6 +13,7 @@ the `ConfigurationBuilder` or by specifying properties in
 . Enable SSL with the `ssl()` method. This implicitly enables SSL if you specify a truststore.
 . Specify the name of the truststore with the `trustStoreFileName()` method.
 . Specify the truststore secret with the `trustStorePassword()` method.
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_configure_clients_trust_server.adoc:15]: I guess this isn't required with "PEM" trust stores right? (at least it works without it)

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/SslConfigurationBuilder.java [hunk 2]
```diff
@@ -256,8 +259,10 @@ public ConfigurationBuilder withProperties(Properties properties) {
       if (typed.containsKey(ConfigurationProperties.TRUST_STORE_FILE_NAME))
          this.trustStoreFileName(typed.getProperty(ConfigurationProperties.TRUST_STORE_FILE_NAME, trustStoreFileName, true));
 
-      if (typed.containsKey(ConfigurationProperties.TRUST_STORE_PATH))
+      if (typed.containsKey(ConfigurationProperties.TRUST_STORE_PATH)) {
+         HOTROD.deprecatedConfigurationProperty(ConfigurationProperties.TRUST_STORE_PATH);
```
Comment (pruivo) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/SslConfigurationBuilder.java:263]: I would move the warning to `trustStorePath()`. Configuring the client programmatically doesn't show any warning.

**Comment by pruivo** [comment]
tested successfuly with:
```
      builder.security().ssl().enable()
            .trustStoreFileName("/tmp/cert/client_truststore.p12")
            .trustStorePassword("clientSecret".toCharArray())
            .trustStoreType("pkcs12");
```
and (generated via `openssl pkcs12 -in client_truststore.p12 -out client.pem`)
```
      builder.security().ssl().enable()
            .trustStoreFileName("/tmp/cert/client.pem")
            .trustStoreType("pem");
```

**Comment by tristantarrant** [comment]
@pruivo I've addressed your comments.

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
integrated! thanks @tristantarrant !

**Pull request merged**